src/main/kotlin/org/jitsi/nlj/PacketInfo.kt

```diff
@@ -88,6 +86,12 @@ open class PacketInfo @JvmOverloads constructor(
      * was an incoming packet and not one created by jvb itself).
      */
     var receivedTime: Long = -1L
+        set(value) {
+            field = value
+            if (ENABLE_TIMELINE && timeline.referenceTime == null) {
```

Comment:
If we have this, does `referenceTime` still need to be public? (How was `referenceTime` getting set before? Or was it not?)

Reply:
`referenceTime` was previously set the first time `addEvent` was called on the timeline.  This change will allow us to have a `referenceTime` earlier than the first `addEvent` - this was happening e.g. for Octo-received packets before I added the Octo receiver queue events in the corresponding JVB PR.

I only made `referenceTime` public so it can be set by `PacketInfo`.  Previously it was private.

Reply:
> referenceTime was previously set the first time addEvent was called on the timeline. This change will allow us to have a referenceTime earlier than the first addEvent - this was happening e.g. for Octo-received packets before I added the Octo receiver queue events in the corresponding JVB PR.

👍 

> I only made referenceTime public so it can be set by PacketInfo. Previously it was private.

Ah, didn't notice it was a different class there.